lib/client.js

```diff
@@ -102,12 +116,16 @@ function rawPostMessage (client, msg, forceReady) {
 //
 function wrappedPostMessage (name, params) {
   const id = nextIdFor('promise')
-  let timeoutId
+  let timeoutId, promiseStartTime
+  const paramsArray = Array.isArray(params) ? params : Object.keys(params)
+  const actions = collateActions(name, paramsArray)
+
   const promise = new Promise((resolve, reject) => {
-    // Time out the promise to ensure it will be garbage collected if nobody responds
-    timeoutId = timeoutReject(reject, name, this, Array.isArray(params) ? params : Object.keys(params))
+    promiseStartTime = window.performance.now()
 
-    pendingPromises[id] = { resolve: resolve, reject: reject }
+    // Time out the promise to ensure it will be garbage collected if nobody responds
+    timeoutId = timeoutReject(this, reject, name, paramsArray)
+    pendingPromises[id] = { resolve, reject }
```

Comment:
Forgive me if I'm a bit naff to ask this, but was the alteration of the structure of this object an intentional refactor?  I'm a bit unclear as to how this variable is used.

Reply:
@token-cjg `pendingPromises[id] = { resolve, reject }` this is es6 object property shorthand.
https://alligator.io/js/object-property-shorthand-es6/

basically, `pendingPromises[id] = { resolve, reject }` is same as `pendingPromises[id] = { resolve: resolve, reject: reject }`